# Project 2: Data Import - Working with Web APIs and JSON (Movies Dataset)

# Project Brief for Self-Coders

Here you´ll have the opportunity to code major parts of Project 2 on your own. If you need any help or inspiration, have a look at the Videos or the Jupyter Notebook with the full code. <br> <br>
Keep in mind that it´s all about __getting the right results/conclusions__. It´s not about finding the identical code. Things can be coded in many different ways. Even if you come to the same conclusions, it´s very unlikely that we have the very same code. 

## Importing Data from JSON files 

1. __Import__ the json files __blockbusters.json__, __blockbusters2.json__, __blockbusters3.json__ and load the datasets into Pandas DataFrames.


In [59]:
import pandas as pd
import json
from dotenv import load_dotenv, dotenv_values

load_dotenv()
config = dotenv_values(".env")

OrderedDict([('TMDB_API_KEY', '81528531e5518f0fabba55759b429b7f'),
             ('TMDB_READ_ACCESS_TOKEN',
              'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI4MTUyODUzMWU1NTE4ZjBmYWJiYTU1NzU5YjQyOWI3ZiIsInN1YiI6IjY0YTlhZTI3YjY4NmI5MDE1MDBmNmRkZSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.XTkt0gWAriPyXAIZhAWVZJOW2o86LKiS6llL0qxLcTg')])

In [41]:
with open('blockbusters.json') as f:
    data = json.load(f)

In [42]:
data

[{'title': 'Avengers: Endgame',
  'id': 299534,
  'revenue': 2797800564,
  'genres': [{'id': 12, 'name': 'Adventure'},
   {'id': 878, 'name': 'Science Fiction'},
   {'id': 28, 'name': 'Action'}],
  'belongs_to_collection': {'id': 86311,
   'name': 'The Avengers Collection',
   'poster_path': '/yFSIUVTCvgYrpalUktulvk3Gi5Y.jpg',
   'backdrop_path': '/zuW6fOiusv4X9nnW3paHGfXcSll.jpg'},
  'runtime': 181},
 {'title': 'Avatar',
  'id': 19995,
  'revenue': 2787965087,
  'genres': [{'id': 28, 'name': 'Action'},
   {'id': 12, 'name': 'Adventure'},
   {'id': 14, 'name': 'Fantasy'},
   {'id': 878, 'name': 'Science Fiction'}],
  'belongs_to_collection': {'id': 87096,
   'name': 'Avatar Collection',
   'poster_path': '/nslJVsO58Etqkk17oXMuVK4gNOF.jpg',
   'backdrop_path': '/8nCr9W7sKus2q9PLbYsnT7iCkuT.jpg'},
  'runtime': 162},
 {'title': 'Star Wars: The Force Awakens',
  'id': 140607,
  'revenue': 2068223624,
  'genres': [{'id': 28, 'name': 'Action'},
   {'id': 12, 'name': 'Adventure'},
   {'id': 8

In [43]:
type(data)

list

In [44]:
len(data)

18

In [45]:
data[0]

{'title': 'Avengers: Endgame',
 'id': 299534,
 'revenue': 2797800564,
 'genres': [{'id': 12, 'name': 'Adventure'},
  {'id': 878, 'name': 'Science Fiction'},
  {'id': 28, 'name': 'Action'}],
 'belongs_to_collection': {'id': 86311,
  'name': 'The Avengers Collection',
  'poster_path': '/yFSIUVTCvgYrpalUktulvk3Gi5Y.jpg',
  'backdrop_path': '/zuW6fOiusv4X9nnW3paHGfXcSll.jpg'},
 'runtime': 181}

In [46]:
df = pd.DataFrame(data)
df

title      id     revenue  \
0                              Avengers: Endgame  299534  2797800564   
1                                         Avatar   19995  2787965087   
2                   Star Wars: The Force Awakens  140607  2068223624   
3                         Avengers: Infinity War  299536  2046239637   
4                                        Titanic     597  1845034188   
5                                 Jurassic World  135397  1671713208   
6                                  The Lion King  420818  1656943394   
7                                   The Avengers   24428  1519557910   
8                                      Furious 7  168259  1506249360   
9                        Avengers: Age of Ultron   99861  1405403694   
10                                 Black Panther  284054  1346739107   
11  Harry Potter and the Deathly Hallows: Part 2   12445  1341511219   
12                      Star Wars: The Last Jedi  181808  1332539889   
13                                     Frozen II  330457  1330764959   
14                Jurassic World: Fallen Kingdom  351286  1303459585   
15                                        Frozen  109445  1274219009   
16                          Beauty and the Beast  321612  1263521126   
17                                 Incredibles 2  260513  1241891456   

                                               genres  \
0   [{'id': 12, 'name': 'Adventure'}, {'id': 878, ...   
1   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
3   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
5   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
6   [{'id': 12, 'name': 'Adventure'}, {'id': 10751...   
7   [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
8   [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
9   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
10  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
11  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...   
12  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   
14  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
15  [{'id': 16, 'name': 'Animation'}, {'id': 12, '...   
16  [{'id': 10751, 'name': 'Family'}, {'id': 14, '...   
17  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   

                                belongs_to_collection  runtime  
0   {'id': 86311, 'name': 'The Avengers Collection...      181  
1   {'id': 87096, 'name': 'Avatar Collection', 'po...      162  
2   {'id': 10, 'name': 'Star Wars Collection', 'po...      136  
3   {'id': 86311, 'name': 'The Avengers Collection...      149  
4                                                None      194  
5   {'id': 328, 'name': 'Jurassic Park Collection'...      124  
6                                                None      118  
7   {'id': 86311, 'name': 'The Avengers Collection...      143  
8   {'id': 9485, 'name': 'The Fast and the Furious...      137  
9   {'id': 86311, 'name': 'The Avengers Collection...      141  
10  {'id': 529892, 'name': 'Black Panther Collecti...      134  
11  {'id': 1241, 'name': 'Harry Potter Collection'...      130  
12  {'id': 10, 'name': 'Star Wars Collection', 'po...      152  
13  {'id': 386382, 'name': 'Frozen Collection', 'p...      104  
14  {'id': 328, 'name': 'Jurassic Park Collection'...      129  
15  {'id': 386382, 'name': 'Frozen Collection', 'p...      102  
16                                               None      129  
17  {'id': 468222, 'name': 'The Incredibles Collec...      118

In [47]:
df = pd.read_json('blockbusters.json')
df

title      id     revenue  \
0                              Avengers: Endgame  299534  2797800564   
1                                         Avatar   19995  2787965087   
2                   Star Wars: The Force Awakens  140607  2068223624   
3                         Avengers: Infinity War  299536  2046239637   
4                                        Titanic     597  1845034188   
5                                 Jurassic World  135397  1671713208   
6                                  The Lion King  420818  1656943394   
7                                   The Avengers   24428  1519557910   
8                                      Furious 7  168259  1506249360   
9                        Avengers: Age of Ultron   99861  1405403694   
10                                 Black Panther  284054  1346739107   
11  Harry Potter and the Deathly Hallows: Part 2   12445  1341511219   
12                      Star Wars: The Last Jedi  181808  1332539889   
13                                     Frozen II  330457  1330764959   
14                Jurassic World: Fallen Kingdom  351286  1303459585   
15                                        Frozen  109445  1274219009   
16                          Beauty and the Beast  321612  1263521126   
17                                 Incredibles 2  260513  1241891456   

                                               genres  \
0   [{'id': 12, 'name': 'Adventure'}, {'id': 878, ...   
1   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
3   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
5   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
6   [{'id': 12, 'name': 'Adventure'}, {'id': 10751...   
7   [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
8   [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
9   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
10  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
11  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...   
12  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   
14  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
15  [{'id': 16, 'name': 'Animation'}, {'id': 12, '...   
16  [{'id': 10751, 'name': 'Family'}, {'id': 14, '...   
17  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   

                                belongs_to_collection  runtime  
0   {'id': 86311, 'name': 'The Avengers Collection...      181  
1   {'id': 87096, 'name': 'Avatar Collection', 'po...      162  
2   {'id': 10, 'name': 'Star Wars Collection', 'po...      136  
3   {'id': 86311, 'name': 'The Avengers Collection...      149  
4                                                None      194  
5   {'id': 328, 'name': 'Jurassic Park Collection'...      124  
6                                                None      118  
7   {'id': 86311, 'name': 'The Avengers Collection...      143  
8   {'id': 9485, 'name': 'The Fast and the Furious...      137  
9   {'id': 86311, 'name': 'The Avengers Collection...      141  
10  {'id': 529892, 'name': 'Black Panther Collecti...      134  
11  {'id': 1241, 'name': 'Harry Potter Collection'...      130  
12  {'id': 10, 'name': 'Star Wars Collection', 'po...      152  
13  {'id': 386382, 'name': 'Frozen Collection', 'p...      104  
14  {'id': 328, 'name': 'Jurassic Park Collection'...      129  
15  {'id': 386382, 'name': 'Frozen Collection', 'p...      102  
16                                               None      129  
17  {'id': 468222, 'name': 'The Incredibles Collec...      118

In [48]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   title                  18 non-null     object
 1   id                     18 non-null     int64 
 2   revenue                18 non-null     int64 
 3   genres                 18 non-null     object
 4   belongs_to_collection  15 non-null     object
 5   runtime                18 non-null     int64 
dtypes: int64(3), object(3)
memory usage: 992.0+ bytes


In [49]:
df.genres[0]

[{'id': 12, 'name': 'Adventure'},
 {'id': 878, 'name': 'Science Fiction'},
 {'id': 28, 'name': 'Action'}]

In [50]:
df.belongs_to_collection[0]

{'id': 86311,
 'name': 'The Avengers Collection',
 'poster_path': '/yFSIUVTCvgYrpalUktulvk3Gi5Y.jpg',
 'backdrop_path': '/zuW6fOiusv4X9nnW3paHGfXcSll.jpg'}

In [51]:
pd.json_normalize(data=data, sep="_")

title      id     revenue  \
0                              Avengers: Endgame  299534  2797800564   
1                                         Avatar   19995  2787965087   
2                   Star Wars: The Force Awakens  140607  2068223624   
3                         Avengers: Infinity War  299536  2046239637   
4                                        Titanic     597  1845034188   
5                                 Jurassic World  135397  1671713208   
6                                  The Lion King  420818  1656943394   
7                                   The Avengers   24428  1519557910   
8                                      Furious 7  168259  1506249360   
9                        Avengers: Age of Ultron   99861  1405403694   
10                                 Black Panther  284054  1346739107   
11  Harry Potter and the Deathly Hallows: Part 2   12445  1341511219   
12                      Star Wars: The Last Jedi  181808  1332539889   
13                                     Frozen II  330457  1330764959   
14                Jurassic World: Fallen Kingdom  351286  1303459585   
15                                        Frozen  109445  1274219009   
16                          Beauty and the Beast  321612  1263521126   
17                                 Incredibles 2  260513  1241891456   

                                               genres  runtime  \
0   [{'id': 12, 'name': 'Adventure'}, {'id': 878, ...      181   
1   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      162   
2   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      136   
3   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...      149   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      194   
5   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      124   
6   [{'id': 12, 'name': 'Adventure'}, {'id': 10751...      118   
7   [{'id': 878, 'name': 'Science Fiction'}, {'id'...      143   
8   [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...      137   
9   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      141   
10  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      134   
11  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...      130   
12  [{'id': 14, 'name': 'Fantasy'}, {'id': 12, 'na...      152   
13  [{'id': 16, 'name': 'Animation'}, {'id': 10751...      104   
14  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      129   
15  [{'id': 16, 'name': 'Animation'}, {'id': 12, '...      102   
16  [{'id': 10751, 'name': 'Family'}, {'id': 14, '...      129   
17  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      118   

    belongs_to_collection_id           belongs_to_collection_name  \
0                    86311.0              The Avengers Collection   
1                    87096.0                    Avatar Collection   
2                       10.0                 Star Wars Collection   
3                    86311.0              The Avengers Collection   
4                        NaN                                  NaN   
5                      328.0             Jurassic Park Collection   
6                        NaN                                  NaN   
7                    86311.0              The Avengers Collection   
8                     9485.0  The Fast and the Furious Collection   
9                    86311.0              The Avengers Collection   
10                  529892.0             Black Panther Collection   
11                    1241.0              Harry Potter Collection   
12                      10.0                 Star Wars Collection   
13                  386382.0                    Frozen Collection   
14                     328.0             Jurassic Park Collection   
15                  386382.0                    Frozen Collection   
16                       NaN                                  NaN   
17                  468222.0           The Incredibles Collection   

   belongs_to_collection_poster_path belongs_to_collection_backdrop_path  \
0   /yFSIUVTCvgYrpalUktulvk3Gi5

In [52]:
pd.json_normalize(data=data, sep="_").genres[0]

[{'id': 12, 'name': 'Adventure'},
 {'id': 878, 'name': 'Science Fiction'},
 {'id': 28, 'name': 'Action'}]

In [53]:
pd.json_normalize(data=data, record_path="genres", meta=["title", "id"], record_prefix="genre_")

genre_id       genre_name                                         title  \
0         12        Adventure                             Avengers: Endgame   
1        878  Science Fiction                             Avengers: Endgame   
2         28           Action                             Avengers: Endgame   
3         28           Action                                        Avatar   
4         12        Adventure                                        Avatar   
5         14          Fantasy                                        Avatar   
6        878  Science Fiction                                        Avatar   
7         28           Action                  Star Wars: The Force Awakens   
8         12        Adventure                  Star Wars: The Force Awakens   
9        878  Science Fiction                  Star Wars: The Force Awakens   
10        14          Fantasy                  Star Wars: The Force Awakens   
11        12        Adventure                        Avengers: Infinity War   
12        28           Action                        Avengers: Infinity War   
13       878  Science Fiction                        Avengers: Infinity War   
14        18            Drama                                       Titanic   
15     10749          Romance                                       Titanic   
16        53         Thriller                                       Titanic   
17        28           Action                                Jurassic World   
18        12        Adventure                                Jurassic World   
19       878  Science Fiction                                Jurassic World   
20        53         Thriller                                Jurassic World   
21        12        Adventure                                 The Lion King   
22     10751           Family                                 The Lion King   
23       878  Science Fiction                                  The Avengers   
24        28           Action                                  The Avengers   
25        12        Adventure                                  The Avengers   
26        28           Action                                     Furious 7   
27        53         Thriller                                     Furious 7   
28        28           Action                       Avengers: Age of Ultron   
29        12        Adventure                       Avengers: Age of Ultron   
30       878  Science Fiction                       Avengers: Age of Ultron   
31        28           Action                                 Black Panther   
32        12        Adventure                                 Black Panther   
33        14          Fantasy                                 Black Panther   
34       878  Science Fiction                                 Black Panther   
35        14          Fantasy  Harry Potter and the Deathly Hallows: Part 2   
36        12        Adventure  Harry Potter and the Deathly Hallows: Part 2   
37        14          Fantasy                      Star Wars: The Last Jedi   
38        12        Adventure                      Star Wars: The Last Jedi   
39       878  Science Fiction                      Star Wars: The Last Jedi   
40        28           Action                      Star Wars: The Last Jedi   
41        16        Animation                                     Frozen II   
42     10751           Family                                     Frozen II   
43        12        Adventure                                     Frozen II   
44        28           Action                Jurassic World: Fallen Kingdom   
45        12        Adventure                Jurassic World: Fallen Kingdom   
46       878  Science Fiction                Jurassic World: Fallen Kingdom   
47        16        Animation                                        Frozen   
48        12        Adventure                                        Frozen   
49     10751           Family                         

## Working with APIs and JSON (Part 1)

2. __Create an account__ on https://www.themoviedb.org/

3. Get your personal __API Key__

4. __API-Request__ (movie module): Load all available information for the movie with __movie id = 140607__ into a Pandas DataFrame. <br> See https://developers.themoviedb.org/3/movies/get-movie-details for more information

In [55]:
import pandas as pd
import requests
pd.options.display.max_columns = 30

In [65]:
movie_id = 140607
api_key = config.get('TMDB_API_KEY')

In [70]:
movie_api = "https://api.themoviedb.org/3/movie/{}?api_key="
movie_api

'https://api.themoviedb.org/3/movie/{}?api_key='

In [71]:
url = movie_api.format(movie_id) + api_key
url

'https://api.themoviedb.org/3/movie/140607?api_key=81528531e5518f0fabba55759b429b7f'

In [72]:
request = requests.get(url)
request

<Response [200]>

In [73]:
data = request.json()

In [74]:
data

{'adult': False,
 'backdrop_path': '/8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg',
 'belongs_to_collection': {'id': 10,
  'name': 'Star Wars Collection',
  'poster_path': '/gq5Wi7i4SF3lo4HHkJasDV95xI9.jpg',
  'backdrop_path': '/d8duYyyC9J5T825Hg7grmaabfxQ.jpg'},
 'budget': 245000000,
 'genres': [{'id': 12, 'name': 'Adventure'},
  {'id': 28, 'name': 'Action'},
  {'id': 878, 'name': 'Science Fiction'}],
 'homepage': 'http://www.starwars.com/films/star-wars-episode-vii',
 'id': 140607,
 'imdb_id': 'tt2488496',
 'original_language': 'en',
 'original_title': 'Star Wars: The Force Awakens',
 'overview': 'Thirty years after defeating the Galactic Empire, Han Solo and his allies face a new threat from the evil Kylo Ren and his army of Stormtroopers.',
 'popularity': 56.11,
 'poster_path': '/wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg',
 'production_companies': [{'id': 1,
   'logo_path': '/o86DbpburjxrqAzEDhXZcyE8pDb.png',
   'name': 'Lucasfilm Ltd.',
   'origin_country': 'US'},
  {'id': 11461,
   'logo_path': '/p9FoE

In [75]:
type(data)

dict

In [77]:
pd.Series(data)

adult                                                                False
backdrop_path                             /8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg
belongs_to_collection    {'id': 10, 'name': 'Star Wars Collection', 'po...
budget                                                           245000000
genres                   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...
homepage                 http://www.starwars.com/films/star-wars-episod...
id                                                                  140607
imdb_id                                                          tt2488496
original_language                                                       en
original_title                                Star Wars: The Force Awakens
overview                 Thirty years after defeating the Galactic Empi...
popularity                                                           56.11
poster_path                               /wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg
production_companies     

In [79]:
df = pd.Series(data).to_frame().T
df

adult                     backdrop_path  \
0  False  /8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg   

                               belongs_to_collection     budget  \
0  {'id': 10, 'name': 'Star Wars Collection', 'po...  245000000   

                                              genres  \
0  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                            homepage      id    imdb_id  \
0  http://www.starwars.com/films/star-wars-episod...  140607  tt2488496   

  original_language                original_title  \
0                en  Star Wars: The Force Awakens   

                                            overview popularity  \
0  Thirty years after defeating the Galactic Empi...      56.11   

                        poster_path  \
0  /wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg   

                                production_companies  \
0  [{'id': 1, 'logo_path': '/o86DbpburjxrqAzEDhXZ...   

                                production_countries release_date     revenue  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   2015-12-15  2068223624   

  runtime                                   spoken_languages    status  \
0     136  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                         tagline                         title  video  \
0  Every generation has a story.  Star Wars: The Force Awakens  False   

  vote_average vote_count  
0        7.295      18139

In [80]:
pd.json_normalize(data=data, sep="_")

adult                     backdrop_path     budget  \
0  False  /8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg  245000000   

                                              genres  \
0  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                            homepage      id    imdb_id  \
0  http://www.starwars.com/films/star-wars-episod...  140607  tt2488496   

  original_language                original_title  \
0                en  Star Wars: The Force Awakens   

                                            overview  popularity  \
0  Thirty years after defeating the Galactic Empi...       56.11   

                        poster_path  \
0  /wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg   

                                production_companies  \
0  [{'id': 1, 'logo_path': '/o86DbpburjxrqAzEDhXZ...   

                                production_countries release_date     revenue  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   2015-12-15  2068223624   

   runtime                                   spoken_languages    status  \
0      136  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                         tagline                         title  video  \
0  Every generation has a story.  Star Wars: The Force Awakens  False   

   vote_average  vote_count  belongs_to_collection_id  \
0         7.295       18139                        10   

  belongs_to_collection_name belongs_to_collection_poster_path  \
0       Star Wars Collection  /gq5Wi7i4SF3lo4HHkJasDV95xI9.jpg   

  belongs_to_collection_backdrop_path  
0    /d8duYyyC9J5T825Hg7grmaabfxQ.jpg

In [81]:
pd.json_normalize(data=data, record_path="genres", meta="title")

id             name                         title
0   12        Adventure  Star Wars: The Force Awakens
1   28           Action  Star Wars: The Force Awakens
2  878  Science Fiction  Star Wars: The Force Awakens

In [82]:
pd.json_normalize(data=data, record_path="production_companies", meta="title")

id                         logo_path            name origin_country  \
0      1  /o86DbpburjxrqAzEDhXZcyE8pDb.png  Lucasfilm Ltd.             US   
1  11461  /p9FoEt5shEKRWRKVIlvFaEmRnun.png       Bad Robot             US   

                          title  
0  Star Wars: The Force Awakens  
1  Star Wars: The Force Awakens

## Working with APIs and JSON (Part 2)

5. __API-Request__ (discover module): Load all movies with __release date between 2020-01-01 and 2020-02-29__ into a Pandas DataFrame. <br>
See https://www.themoviedb.org/documentation/api/discover and https://developers.themoviedb.org/3/discover/movie-discover for more information.

##  Importing and Saving the Movies Dataset (Best Practice)

6. __API-Request__ (movie module): Load all available information for the movies with movie id = [__299534, 19995, 140607, 299536, 597, 135397, 420818, 24428, 168259, 99861, 284054, 12445, 181808, 330457, 351286, 109445, 321612, 260513__] into a Pandas DataFrame and __save the dataset in a local json file__.

# +++++++++ See some Hints below +++++++++++++

# ++++++++++++++ Hints +++++++++++++++++++++

__Hints for 1.__ <br>
To load json files you can use 

In [54]:
with open("filename.json") as f:
    data = json.load(f)

FileNotFoundError: [Errno 2] No such file or directory: 'filename.json'

and 

In [ ]:
pd.DataFrame(data), pd.read_json(filename.json), pd.json_normalize(data)

the json files have the following orientation (important when using pd.read_json()):
- blockbusters.json -> record
- blockbusters2.json -> column
- blockbusters3.json -> split 

__Hints for 4., 5., 6.__<br>
Make API GET-requests with the library requests (import requests):

In [ ]:
data = requests.get(url).json()

__Hints for 4. and 6.,__ <br> url structure for movie module:

"https://api.themoviedb.org/3/movie/insert_movie_id?api_key=insert_api_key" (replace "insert_movie_id" with movie id and "insert_api_key" with your personal api-key)

__Hints for 5.__<br>
url structure for discover module:

"https://api.themoviedb.org/3/discover/movie?api_key=insert_api_key&query1&query2..." (replace "insert_api_key" with your personal api-key and add appropriate queries)